instrumentation/spymemcached-2.12/src/main/java/io/opentelemetry/instrumentation/auto/spymemcached/CompletionListener.java

```diff
@@ -28,29 +26,27 @@ public CompletionListener(MemcachedConnection connection, String methodName) {
   }
 
   protected void closeAsyncSpan(T future) {
-    try (Scope ignored = currentContextWith(span)) {
-      try {
-        processResult(span, future);
-      } catch (CancellationException e) {
+    try {
+      processResult(span, future);
```

Comment:
What if `processResult` creates new spans? Will their parent change now?

Reply:
`processResult` is only used to populate span attributes, it doesn't make callbacks into the user code, so I think this is ok